builder/comp-builder.nix

```diff
@@ -184,11 +226,10 @@ stdenv.mkDerivation ({
     config = component;
     inherit configFiles executableToolDepends cleanSrc exeName;
     env = shellWrappers;
-
-    # The directory containing the haddock documentation.
-    # `null' if no haddock documentation was built.
-    haddockDir = if doHaddock' then "${docdir drv.doc}/html" else null;
     profiled = self (drvArgs // { enableLibraryProfiling = true; });
+  } // lib.optionalAttrs (haskellLib.isLibrary componentId) {
+    inherit haddock;
+    inherit (haddock) doc haddockDir;
```

Comment:
Yes, this keeps the interface the same so `shellFor` and any user code using `.doc` should keep working.

Reply:
I guess I'm wondering if anything weird will happen with things that expect it to be a multiple-outputs output. I think it might actually just work nicely, though.

Reply:
Do we need `haddockDir` as well?

Reply:
It's a bit odd but I think it is used by the `hoogle` code we borrow from `nixpkgs` in shellFor.  Definitely ripe for refactoring (in another PR though).